.github/workflows/run_tests.yml

```
@@ -124,3 +124,22 @@ jobs:
           pip install numpy==1.24.0 '.[test-core,plots]'
       - name: Test with pytest
         run: pytest --durations=20 --import-mode=append
+
+  run_ypy:
+    # Run mypy on the latest Python version
+    name: run mypy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          # Use "eager" update strategy in case cached dependencies are outdated
+          # Using regular install NOT editable install: see GH #3020
+          pip install --upgrade --upgrade-strategy eager '.[test-core,plots]'
```

This can be simplified a little as we're not caching this job:

```suggestion
          pip install '.[test-core]'
```